scrape/scrape.go

```diff
@@ -619,7 +620,7 @@ func verifyLabelLimits(lset labels.Labels, limits *labelLimits) error {
 
 	met := lset.Get(labels.MetricName)
 	if limits.labelLimit > 0 {
-		nbLabels := len(lset)
+		nbLabels := lset.Len()
```

Comment:
Regarding the cost of len.
As this happens on every append it might really be worth making it cheaper.

Reply:
Also as an idea for future optimization.
How about we provide the limits to your new label processing code and validate labels while they are being written instead of iterating another time afterwards?

Reply:
On my reading of the code it does not happen on every append. It is called from the branch if the series is not in cache, i.e. if it has never been seen before.

https://github.com/prometheus/prometheus/pull/10991/commits/5a472f285ae81523ce28c8b0f52ea80c69ea227e#diff-4535d6172e0a9ab4e0879d71ef67ab69023f6eda9b44100343a6afa04fc8925fR1559